src/plugins/janus_sip.c

```diff
@@ -1801,6 +1801,95 @@ static void janus_sip_sofia_logger(void *stream, char const *fmt, va_list ap) {
 	}
 }
 
+static void janus_sip_sofia_logger_siptrace_single_log_callback(void *stream, char const *fmt, va_list ap) {
+	/* Since the fmt format string in the current Sofia SIP tport_log_msg function implementation is just "%s\n",
+	 * it's more efficient to directly work with the siptrace buffer. */
+	char *buffer = va_arg(ap, char *);
+
+	/* Check if this is a message we need */
+	if(buffer[0] == 's' && buffer[1] == 'e' && buffer[2] == 'n' && buffer[3] == 'd' && buffer[4] == ' ') {
+		/* An outgoing message is going to be logged, prepare for that */
+		int length = atoi(&buffer[5]);
+		JANUS_LOG(LOG_HUGE, "Intercepting message (%d bytes)\n", length);
+		/* Skip the stamp line */
+		buffer = strchr(buffer, '\n');
+		if(!buffer) {
+			JANUS_LOG(LOG_WARN, "No newline after the stamp in this message, dropping it...\n");
+			return;
+		}
+		buffer++;
+		/* Skip the separator line */
+		buffer = strchr(buffer, '\n');
+		if(!buffer) {
+			JANUS_LOG(LOG_WARN, "No newline after the separator in this message, dropping it...\n");
+			return;
+		}
+		buffer++;
+		/* If this is an OPTIONS, we don't care: drop it */
+		if(strstr(buffer, "OPTIONS") == buffer)
+			return;
```

Comment:
Why? This callback is meant to pass _ALL_ SIP messages to event handlers, as long as they belong to a session. This OPTIONS may be part of a dialog.

Reply:
This logic was adapted from the original `janus_sip_sofia_logger` handler function [here](https://github.com/meetecho/janus-gateway/blob/ea2799743d49938077c8267d85af1cf79777d3c5/src/plugins/janus_sip.c#L1778-L1781). I included it here to maintain consistency with the original implementation. If you approve of having different logic in logger handlers, I can remove it.

Reply:
@lminiero waiting for your confirmation to remove this condition to drop OPTIONS messages.
However, it’s worth noting that in my cases, the PBX-es sends numerous automated without any meaning OPTIONS messages (roughly every 2-5 seconds) within each SIP session, likely to maintain NAT ports. This was probable the reason they were originally dropped in the logger handler.

Reply:
I completely agree with you. Therefore, I will leave the current logic unchanged for now.